### PR TITLE
added 'flask api' with function endpoints. Unittest pending...

### PR DESCRIPTION
**Pull request by verasraul**
### **Added Flask API**

- Created new app.py file (flask module).

- Created 3 route/endpoints:

 1.  Home page:
`@app.route("/")`
`def home():`
    `return "Welcome Home"`

Screen Shot 2019-04-25 at 11 41 49 PM

   2.  Divider page:
`@app.route("/divider")`
`def divider():`
    `return str((dividerFunction1.divider(10,2)))`

Screen Shot 2019-04-25 at 11 45 09 PM

   3.  Addition page:
`@app.route("/addition")`
`def addition():`
    `return str((additionFunction1.addition(5,5)))`

Screen Shot 2019-04-25 at 11 50 59 PM

@agaro1121

### CODE REVIEW

File: PracticeCode/AntProjects/DividerFunc/app.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+app = Flask(__name__)
+
+
+@app.route("/")
+def home():
+    return "Welcome Home"
+
+@app.route("/divider")
+def divider():
+    return str((dividerFunction1.divider(10,2)))
```
Comment (agaro1121) [inline PracticeCode/AntProjects/DividerFunc/app.py]: make this a `POST` so clients can pass in arguments

**Comment by agaro1121** [comment]
Great start!

Some **TODO** items:

[ ] All endpoints **MUST** return JSON in the form of:
```
{
  "result": 0.0
}
```
[ ] Division endpoint **MUST** accept arguments - you can do this via `POST`
  - This endpoint has an edge case. Dividing by zero results in an error. Provide the ability for clients to enable debugging via query parameter. See example call below:
`POST http://localhost:5000/divider?debug=true`
body: 
```
{
  "dividend": 1.0,
  "divisor": 0
}
```
**PLEASE NOTE:** I should be able to pass those arguments in any order and still get the same result i.e 
```
{
  "divisor": 0,
  "dividend": 1.0
}
```
result:
```
{
  "error": {
    "reason": "Cannot compute",
    "debug": "Cannot divide by zero"
  }
}
```

If debug is not passed in or it's `false`, then result **MUST** look like the following:
```
{
  "error": {
    "reason": "Cannot compute"
  }
}

```
[ ] Addition endpoint **MUST** accept arguments - you can do this via `POST`
`POST http://localhost:5000/addition`
body:
```
{
  "number1": 1.0,
  "number2": 2.0 
}
```
result:
```
{
  "result": 3.0
}
```


**Last note:**
- These requirements may not all be necessary such as the `debug` query parameter, but this is a contrived example to expose you to a large breadth of http practice
- Please use a proper json library to do all json manipulation such as [this](https://docs.python.org/3/library/json.html)

File: PracticeCode/AntProjects/DividerFunc/app.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        dividend = content["dividend"]
+        try:
+            return jsonify({"answer": (dividerFunction1.divider(dividend, divisor))})
+        except ZeroDivisionError:
+            return jsonify({
+                "error": {
+                    "reason": "Cannot compute",
+                    "debug": "Cannot divide by zero"
+                }
+            })
```
Comment (agaro1121) [inline PracticeCode/AntProjects/DividerFunc/app.py]: duplicate code. Please refactor

File: PracticeCode/AntProjects/DividerFunc/app.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+@app.route("/divider", methods=['POST', 'GET'])
+def divider():
+    checkreq = request.method
+    if checkreq == 'GET':
+        divisor = float(request.args.get('divisor'))
+        dividend = float(request.args.get('dividend'))
+        print(divisor)
+        print(dividend)
```
Comment (agaro1121) [inline PracticeCode/AntProjects/DividerFunc/app.py]: remove print statements

File: PracticeCode/AntProjects/DividerFunc/app.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                }
+            })
+
+
+@app.route("/addition", methods=['POST'])
+def adder():
+    content = request.get_json()
+    number1 = content['number1']
+    number2 = content["number2"]
+    return jsonify({"sum": (additionFunction1.addition(number1, number2))})
```
Comment (agaro1121) [inline PracticeCode/AntProjects/DividerFunc/app.py]: The return type above has `result` as the main part of the body. This one has `sum`.
This makes the API inconsistent and won't be very obvious for clients.
Standardize on `result` and extract that common code out

If asked to add a `subtraction` function, how easy would it be with/without the above suggestion?

File: PracticeCode/AntProjects/DividerFunc/app.py (hunk 3 above)
Comment (verasraul) [reply PracticeCode/AntProjects/DividerFunc/app.py]: Print statements removed

File: PracticeCode/AntProjects/DividerFunc/app.py (hunk 2 above)
Comment (verasraul) [reply PracticeCode/AntProjects/DividerFunc/app.py]: Duplicate code removed.

File: PracticeCode/AntProjects/DividerFunc/app.py (hunk 4 above)
Comment (verasraul) [reply PracticeCode/AntProjects/DividerFunc/app.py]: Consistency and standardization across API was modified/adjusted.

**Review by agaro1121 — APPROVED** [review]

**Pull request merged**